proposals/2020-07-01_aws-kms-keyring-redesign/change.md

```diff
@@ -0,0 +1,340 @@
+that initializes AWS SDK clients.
+AWS SDK clients are supplied to the AWS KMS keyring at the keyring’s API runtime.
+Client suppliers maintain logic for limiting communication to specific AWS regions and caching AWS SDK clients.
+
+To reduce the scope of the AWS KMS keyring,
+we are moving the keyring’s functionality into multiple smaller-scoped keyrings.
+These smaller-scoped keyrings are configured by _keyring-producing operations_.
+Depending on the implementation, keyring-producing operations can be thought of as factory methods/builders.
+They initialize keyrings.
+
+Keyring-producing operations act as the customer-facing APIs.
+Like client suppliers, they initialize AWS SDK clients.
+However, AWS SDK clients are now initialized before a keyring is configured.
+Keyring-producing operations also handle the load of manually initializing multiple, smaller-scoped keyrings.
+They initialize all keyrings required to meet a specific customer use case.
+We refer to the keyrings that keyring-producing operations produce as _derived keyrings_.
+If a use case involves multiple AWS regions or multiple key names,
+the keyring-producing operation will initialize a [multi-keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/multi-keyring.md)
+of smaller-scoped keyrings.
+
+In summary, rather than having a single keyring maintain all AWS KMS logic,
+logic is broken up into multiple smaller-scoped keyrings.
+Rather than support multiple keys/AWS regions in a single AWS KMS keyring,
+we use keyring-producing operations to configure a [multi-keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/multi-keyring.md)
+of these smaller-scoped keyrings.
+The smaller-scoped AWS KMS keyrings become the basic building blocks that keyring-producing operations use to satisfy different customer use cases.
+
+From a customer standpoint, naming will be clearer.
+Most customers will not directly interact with the basic building blocks
+and will only interact with the keyring-producing operations.
+However, more advanced customers can directly interact with the smaller-scoped keyrings
+and write their own keyring-producing operations.
+This gives advanced customers more flexibility and reduces the need to fork ESDK code in order to meet a specific use case.
+
+## Reference-level Explanation
+
+### AWS KMS symmetric keyring
+
+**Initialization**
+
+On keyring initialization, an AWS KMS symmetric keyring MUST define the following:
+
+- [Key Name](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#key-names)
+- AWS SDK KMS service client (AWS SDK client)
+
+On keyring initialization, an AWS KMS symmetric keyring MAY define the following:
```

Comment:
Should this MAY stand?
For me the issue is "MAY define",
if it was "MAY receive"...

Reply:
Will rephrase based on the above.